server/storage/mvcc/key_index.go

```diff
@@ -116,6 +116,29 @@ func (ki *keyIndex) restore(lg *zap.Logger, created, modified Revision, ver int6
 	keysGauge.Inc()
 }
 
+// restoreTombstone is only used to build index when there is no previous revision.
+// Since there is no historical information, CreateRevision always is empty.
+func (ki *keyIndex) restoreTombstone(lg *zap.Logger, main, sub int64) {
```

Comment:
Apart of not increasing `keyGauge` this function doesn't differs from `restore`. Maybe we could combine them as they represent the same process just with different parameters? 

Think it should be equivalent to `restore(lg, 0, Revision{main, sub}, 1)`

Reply:
hi @serathius @ahrtr do you mean by this?

```go
func (ki *keyIndex) restoreTombstone(lg *zap.Logger, main, sub int64) {
     ki.restore(lg, Revision{}, Revsion{main, sub}, 1)
     ki.generations = append(ki.generations, generation{})
     keyGauge.Dec()
}
```

I was thinking about combining the following functions instead of introducing new function. What do you think about it?

```go
ki.restore(lg, Revision{}, Revision{main-1, sub}, 0) just introduces previous revision and then compact it
ki.tombstone(lg, main, sub)
ki.compact(lg, main, map[Revision]struct{}{})
```

Reply:
> hi @serathius @ahrtr do you mean by this?
> 
> ```go
> func (ki *keyIndex) restoreTombstone(lg *zap.Logger, main, sub int64) {
>      ki.restore(lg, Revision{}, Revsion{main, sub}, 1)
>      ki.generations = append(ki.generations, generation{})
>      keyGauge.Dec()
> }
> ```

It's also an valid option.